File: app/src/main/java/br/com/pauloc/mercadin/repositories/ProdutoRepositorio.java
package br.com.pauloc.mercadin.repositories;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.SQLException;
import android.database.sqlite.SQLiteDatabase;
import android.provider.MediaStore;

import java.util.ArrayList;

import br.com.pauloc.mercadin.DB.DataBaseSQLHelper;
import br.com.pauloc.mercadin.model.Produto;

import static android.provider.BaseColumns._ID;
import static br.com.pauloc.mercadin.DB.DataBaseSQLHelper.COLUNA_DESCRICAO;
import static br.com.pauloc.mercadin.DB.DataBaseSQLHelper.COLUNA_PRODUTO_CATEGORIA;
import static br.com.pauloc.mercadin.DB.DataBaseSQLHelper.COLUNA_PRODUTO_MARCA;
import static br.com.pauloc.mercadin.DB.DataBaseSQLHelper.COLUNA_PRODUTO_STATUS;
import static br.com.pauloc.mercadin.DB.DataBaseSQLHelper.COLUNA_QNT;
import static br.com.pauloc.mercadin.DB.DataBaseSQLHelper.COLUNA_VALIDADE;
import static br.com.pauloc.mercadin.DB.DataBaseSQLHelper.COLUNA_VALOR;
import static br.com.pauloc.mercadin.DB.DataBaseSQLHelper.TABELA_PRODUTO;

public class ProdutoRepositorio {
    private static String BASEDADOS_TABELA = TABELA_PRODUTO;
    private DataBaseSQLHelper helper;
    private Context context;
    private SQLiteDatabase database;


    public ProdutoRepositorio(Context context) {
        this.context = context;

    }


    public void close() {
        helper.close();
    }

    public long inserir(Produto produto) {
        SQLiteDatabase db = helper.getWritableDatabase();
        ContentValues cv = new ContentValues();
        cv.put(COLUNA_DESCRICAO, produto.descricao);
        cv.put(COLUNA_VALOR, produto.valor);
        cv.put(COLUNA_VALIDADE, produto.validade);
        cv.put(COLUNA_QNT, produto.qntItens);
        cv.put(COLUNA_PRODUTO_MARCA, produto.marca);
        cv.put(COLUNA_PRODUTO_CATEGORIA, produto.categoria);
        cv.put(COLUNA_PRODUTO_STATUS, produto.status);
        long id = db.insert(TABELA_PRODUTO, null, cv);
        if (id != -1) {
            produto.id = id;

        }
        db.close();
        return id;
    }

    public ProdutoRepositorio open() throws SQLException {
        helper = new DataBaseSQLHelper(context);

        database = helper.getWritableDatabase();
        return this;
    }

    public int update(Produto produto) {
        ContentValues contentValues = new ContentValues();
        contentValues.put(COLUNA_DESCRICAO, produto.getDescricao());
        contentValues.put(COLUNA_QNT, produto.getQntItens());
        contentValues.put(COLUNA_PRODUTO_MARCA, produto.getMarca());
        contentValues.put(COLUNA_PRODUTO_CATEGORIA, produto.getCategoria());
        contentValues.put(COLUNA_VALIDADE, produto.getValidade());
        contentValues.put(COLUNA_VALOR, produto.getValor());

        return database.update(
                BASEDADOS_TABELA,
                contentValues,
                _ID + "= '" + produto.getId() + "'",
                null
        );
    }

    public int delete(int id) {
        return database.delete(TABELA_PRODUTO, _ID + " = '" + id + "'", null);
    }

    public int insert(Produto produto) {
        ContentValues contentValues = new ContentValues();
        contentValues.put(COLUNA_DESCRICAO, produto.getDescricao());
        contentValues.put(COLUNA_QNT, produto.getQntItens());
        contentValues.put(COLUNA_VALIDADE, produto.getValidade());
        contentValues.put(COLUNA_VALOR, produto.getValor());

        return (int) database.insert(BASEDADOS_TABELA, null, contentValues);
    }

    public ArrayList<Produto> query() {
        ArrayList<Produto> arrayList = new ArrayList<Produto>();

        Cursor cursor = database.rawQuery(
                "select * from produto where status =? ORDER BY descricao asc",
                new String[] {"0"});

//        Cursor cursor = database.query(
//                BASEDADOS_TABELA,
//                null,
//                null,
//                null,
//                null,
//                null,
//                MediaStore.Audio.Playlists.Members._ID + " DESC",
//                null
//        );

        cursor.moveToFirst();

        Produto produto;

        if (cursor.getCount() > 0) {
            do {
                produto = new Produto();

                produto.setId(cursor.getInt(cursor.getColumnIndexOrThrow(MediaStore.Audio.Playlists.Members._ID)));
                produto.setDescricao(cursor.getString(cursor.getColumnIndexOrThrow(COLUNA_DESCRICAO)));
                produto.setQntItens(cursor.getInt(cursor.getColumnIndexOrThrow(COLUNA_QNT)));
                produto.setMarca(cursor.getString(cursor.getColumnIndexOrThrow(COLUNA_PRODUTO_MARCA)));
                produto.setCategoria(cursor.getString(cursor.getColumnIndexOrThrow(COLUNA_PRODUTO_CATEGORIA)));
                produto.setValor(cursor.getString(cursor.getColumnIndexOrThrow(COLUNA_VALOR)));
                produto.setValidade(cursor.getString(cursor.getColumnIndexOrThrow(COLUNA_VALIDADE)));
                produto.setStatus(cursor.getInt(cursor.getColumnIndexOrThrow(COLUNA_PRODUTO_STATUS)));

                arrayList.add(produto);

                cursor.moveToNext();
            } while (!cursor.isAfterLast());
        }
        cursor.close();

        return arrayList;
    }


}

File: app/src/main/java/br/com/pauloc/mercadin/repositories/ProdutoRepositorio.kt
package br.com.pauloc.mercadin.repositories

import android.content.ContentValues
import android.content.Context
import android.database.SQLException
import android.database.sqlite.SQLiteDatabase
import android.provider.BaseColumns
import android.provider.ContactsContract
import android.provider.MediaStore
import br.com.pauloc.mercadin.DB.DataBaseSQLHelper
import br.com.pauloc.mercadin.model.Produto
import java.util.*

class ProdutoRepositorio(private val context: Context) {
    private var helper: DataBaseSQLHelper? = null
    private var database: SQLiteDatabase? = null
    fun close() {
        helper!!.close()
    }

    fun inserir(produto: Produto): Long {
        val db = helper!!.writableDatabase
        val cv = ContentValues()
        cv.put(DataBaseSQLHelper.COLUNA_DESCRICAO, produto.descricao)
        cv.put(DataBaseSQLHelper.COLUNA_VALOR, produto.valor)
        cv.put(DataBaseSQLHelper.COLUNA_VALIDADE, produto.validade)
        cv.put(DataBaseSQLHelper.COLUNA_QNT, produto.qntItens)
        cv.put(DataBaseSQLHelper.COLUNA_PRODUTO_MARCA, produto.marca)
        cv.put(DataBaseSQLHelper.COLUNA_PRODUTO_CATEGORIA, produto.categoria)
        cv.put(DataBaseSQLHelper.COLUNA_PRODUTO_STATUS, produto.status)
        val id = db.insert(DataBaseSQLHelper.TABELA_PRODUTO, null, cv)
        if (id != -1L) {
            produto.id = id
        }
        db.close()
        return id
    }

    fun open(): ProdutoRepositorio {
        helper = DataBaseSQLHelper(context)
        database = helper!!.writableDatabase
        return this
    }

    fun update(produto: Produto): Int {
        val contentValues = ContentValues()
        contentValues.put(DataBaseSQLHelper.COLUNA_DESCRICAO, produto.descricao)
        contentValues.put(DataBaseSQLHelper.COLUNA_QNT, produto.qntItens)
        contentValues.put(DataBaseSQLHelper.COLUNA_PRODUTO_MARCA, produto.marca)
        contentValues.put(DataBaseSQLHelper.COLUNA_PRODUTO_CATEGORIA, produto.categoria)
        contentValues.put(DataBaseSQLHelper.COLUNA_VALIDADE, produto.validade)
        contentValues.put(DataBaseSQLHelper.COLUNA_VALOR, produto.valor)
        return database!!.update(
                BASEDADOS_TABELA,
                contentValues,
                BaseColumns._ID + "= '" + produto.id + "'",
                null
        )
    }

    fun updateStatus(produto: Produto): Int {
        val contentValues = ContentValues()
        contentValues.put(DataBaseSQLHelper.COLUNA_DESCRICAO, produto.descricao)
        contentValues.put(DataBaseSQLHelper.COLUNA_QNT, produto.qntItens)
        contentValues.put(DataBaseSQLHelper.COLUNA_PRODUTO_MARCA, produto.marca)
        contentValues.put(DataBaseSQLHelper.COLUNA_PRODUTO_CATEGORIA, produto.categoria)
        contentValues.put(DataBaseSQLHelper.COLUNA_VALIDADE, produto.validade)
        contentValues.put(DataBaseSQLHelper.COLUNA_VALOR, produto.valor)
        contentValues.put(DataBaseSQLHelper.COLUNA_PRODUTO_STATUS, 0)
        return database!!.update(
                BASEDADOS_TABELA,
                contentValues,
                BaseColumns._ID + "= '" + produto.id + "'",
                null
        )
    }

    fun updateStatusAll(): Int{
        val contentValues = ContentValues()
        contentValues.put(DataBaseSQLHelper.COLUNA_PRODUTO_STATUS, 0);
        return database!!.update(
                BASEDADOS_TABELA,
                contentValues,
                null,
                null
        )
    }

    fun updateStatus2(id: Int): Int{
        val contentValues = ContentValues()
        contentValues.put(DataBaseSQLHelper.COLUNA_PRODUTO_STATUS, 1);
        return database!!.update(
                BASEDADOS_TABELA,
                contentValues,
                BaseColumns._ID + "= '" + id + "'",
                null
        )
    }

    fun delete(id: Int): Int {
        return database!!.delete(DataBaseSQLHelper.TABELA_PRODUTO, BaseColumns._ID + " = '" + id + "'", null)
    }

    fun insert(produto: Produto): Int {
        val contentValues = ContentValues()
        contentValues.put(DataBaseSQLHelper.COLUNA_DESCRICAO, produto.descricao)
        contentValues.put(DataBaseSQLHelper.COLUNA_QNT, produto.qntItens)
        contentValues.put(DataBaseSQLHelper.COLUNA_VALIDADE, produto.validade)
        contentValues.put(DataBaseSQLHelper.COLUNA_VALOR, produto.valor)
        return database!!.insert(BASEDADOS_TABELA, null, contentValues).toInt()
    }

    fun query(): ArrayList<Produto> {
        val arrayList = ArrayList<Produto>()
        val cursor = database!!.rawQuery(
                "select * from produto where status =? ORDER BY descricao asc", arrayOf("0"))

//        Cursor cursor = database.query(
//                BASEDADOS_TABELA,
//                null,
//                null,
//                null,
//                null,
//                null,
//                MediaStore.Audio.Playlists.Members._ID + " DESC",
//                null
//        );
        cursor.moveToFirst()
        var produto: Produto
        if (cursor.count > 0) {
            do {
                produto = Produto()
                produto.id = cursor.getInt(cursor.getColumnIndexOrThrow(MediaStore.Audio.Playlists.Members._ID)).toLong()
                produto.descricao = cursor.getString(cursor.getColumnIndexOrThrow(DataBaseSQLHelper.COLUNA_DESCRICAO))
                produto.qntItens = cursor.getInt(cursor.getColumnIndexOrThrow(DataBaseSQLHelper.COLUNA_QNT))
                produto.marca = cursor.getString(cursor.getColumnIndexOrThrow(DataBaseSQLHelper.COLUNA_PRODUTO_MARCA))
                produto.categoria = cursor.getString(cursor.getColumnIndexOrThrow(DataBaseSQLHelper.COLUNA_PRODUTO_CATEGORIA))
                produto.valor = cursor.getString(cursor.getColumnIndexOrThrow(DataBaseSQLHelper.COLUNA_VALOR))
                produto.validade = cursor.getString(cursor.getColumnIndexOrThrow(DataBaseSQLHelper.COLUNA_VALIDADE))
                produto.status = cursor.getInt(cursor.getColumnIndexOrThrow(DataBaseSQLHelper.COLUNA_PRODUTO_STATUS))
                arrayList.add(produto)
                cursor.moveToNext()
            } while (!cursor.isAfterLast)
        }
        cursor.close()
        return arrayList
    }

    companion object {
        private const val BASEDADOS_TABELA = DataBaseSQLHelper.TABELA_PRODUTO
    }
}